enterprise/backend/test/metabase_enterprise/advanced_permissions/common_test.clj

```diff
@@ -169,22 +169,112 @@
+                string schema, it should return the table"
+        (with-all-users-data-perms {db-id {:data       {:schemas :all :native :write}
+                                           :data-model {:schemas {"" {(u/the-id t1) :all
+                                                                      (u/the-id t3) :none}}}}}
+          (is (= ["t1"]
+                 (map :name (mt/user-http-request :rasta :get 200 (format "database/%d/schema/" db-id)
+                                                  :include_editable_data_model true)))))))))
+
+(deftest get-schemas-with-advanced-perms-test
+  (testing "Permissions: We can verify include_editable_data_model flag works for the `/:id/:schemas` endpoint"
+    (mt/with-temp* [Database [{db-id :id}]
+                    Table    [t1 {:db_id db-id, :schema "schema1", :name "t1"}]
+                    Table    [_t2 {:db_id db-id, :schema "schema2"}]
+                    Table    [t3 {:db_id db-id, :schema "schema1", :name "t3"}]]
+      (with-all-users-data-perms {db-id {:data       {:schemas :all :native :write}
+                                         :data-model {:schemas :all}}}
+        (perms/revoke-data-perms! (perms-group/all-users) db-id)
+        (testing "If data permissions are revoked, it should be a 403"
+          (is (= "You don't have permissions to do that."
+                 (mt/user-http-request :rasta :get 403 (format "database/%d/schemas" db-id)))))
+        (testing "If include_editable_data_model=true and data permissions are revoked, it should return values"
+          (is (= ["schema1" "schema2"]
+                 (mt/user-http-request :rasta :get 200 (format "database/%d/schemas" db-id)
+                                       :include_editable_data_model true))))
+        (testing "If include_editable_data_model=true and a non-admin does not have data model perms, it should return []"
+          (with-all-users-data-perms {db-id {:data       {:schemas :all :native :write}
+                                             :data-model {:schemas :none}}}
+            (is (= []
+                   (mt/user-http-request :rasta :get 200 (format "database/%d/schemas" db-id)
+                                         :include_editable_data_model true)))))
+
+        (testing "If include_editable_data_model=true and a non-admin has data model perms for a schema,
+                  it should return the schema"
+          (with-all-users-data-perms {db-id {:data       {:schemas :all :native :write}
+                                             :data-model {:schemas {"schema1" :all}}}}
+            (is (= ["schema1"]
+                   (mt/user-http-request :rasta :get 200 (format "database/%d/schemas" db-id)
+                                         :include_editable_data_model true)))))
+
+        (testing "If include_editable_data_model=true and a non-admin has data model perms for a single table in a schema,
+                  it should return the schema"
```

Comment:
This test would fail if not for this change:
https://github.com/metabase/metabase/pull/30352/files#diff-2ea0158f55529133265c09efd01fd6172a26e6b6fc1fcc99c9fdd8b2fcc9ce66R67